plugins/application/loki/pkg/lib/loki.go

```diff
@@ -0,0 +1,51 @@
+package lib
+
+import (
+	"time"
+	"fmt"
+
+	"github.com/infrawatch/apputils/connector"
+	"github.com/infrawatch/sg-core/pkg/data"
+)
+
+type LokiConfig struct {
+	Connection  string `validate:"required"`
+	BatchSize   int
+	MaxWaitTime int
+}
+
+// Creates labels used by Loki. Discards everything, that isn't
```

Comment:
Would this not leave out any important information? I'm not as familiar with the structure of incoming logs, but I wonder if it would be better to just convert everything to a string type

Reply:
The current log format sent by our rsyslog configuration can be seen [here](https://github.com/infrawatch/sg-core/blob/8b6e7586c7c396cf81d57972a3d346d18296c79c/plugins/transport/dummy-logs/main.go#L19). My thinking was, that everything we receive is a string, so this in fact shouldn't discard anything. I actually think, we might have too many labels.

Reply:
If in the future, a different log reporter uses an Int value for one of the labels or if one of the handlers adds a label that has a a value of something other than a string, it may lead to confusion as to why they aren't showing up. You could just use a similar function to Martin's assimilateMap func: https://github.com/infrawatch/sg-core/blob/8b6e7586c7c396cf81d57972a3d346d18296c79c/plugins/application/alertmanager/pkg/lib/utils.go#L16

Reply:
If we wanted to omit some of the labels, that would be the responsibility of the log handler to filter some out

Reply:
I added the conversion with Martin's assimilateMap function.